lnwallet/wallet.go

```diff
@@ -507,6 +528,18 @@ func (l *LightningWallet) handleFundingReserveRequest(req *InitFundingReserveMsg
 
 		localFundingAmt = fundingIntent.LocalFundingAmt()
 		remoteFundingAmt = fundingIntent.RemoteFundingAmt()
+
+	} else {
+		l.intentMtx.Lock()
+		// Alternatively, if we've just received an inbound funding
+		// request that we have a registered shim intent to, then we'll
+		// obtain the backing intent now. In this case, we're doing a
+		// special funding workflow that allows more advanced
+		// constructions such as channel factories to be instantiated.
+		if shimIntent, ok := l.fundingIntents[req.PendingChanID]; ok {
```

Comment:
I don't get this. Isn't the point of the canned assembler that you can use it to fetch such `pre-created` intents? Now there are two paths to achieve this: 1) pre-register an intent 2) create a canned assembler that when called will return the pre-created intent.

It should be enough with one of them?

Reply:
No we need both. Please see how this is exposed on the RPC interface in the PR that builds upon this one. In the context of your example: #1 is done by the _responder_ to an external funding event, while your #2 is done by the _initiator_ of the external funding event.

Reply:
The `initiator` field is used to make the local/remote funding amount attributes consistent. See `lnwallet/wallet.go` for how the attributes/methods are used within the wallet. They're a part of the main intent interface.